logger/index.ts

```diff
@@ -1,45 +1,93 @@
 import format from "@arcjet/sprintf";
 
-export type LogLevel = "DEBUG" | "LOG" | "WARN" | "ERROR";
-
-function getEnvLogLevel(): LogLevel | undefined {
-  const level = process.env["ARCJET_LOG_LEVEL"];
-  switch (level) {
-    case "DEBUG":
-    case "LOG":
-    case "WARN":
-    case "ERROR":
-      return level;
-    default:
-      return undefined;
+function bigintReplacer(key: string, value: unknown) {
+  if (typeof value === "bigint") {
+    return "[BigInt]";
+  }
+
+  return value;
+}
+
+// TODO: Deduplicate this and sprintf implementation
+function tryStringify(o: unknown) {
+  try {
+    return JSON.stringify(o, bigintReplacer);
+  } catch (e) {
+    return "[Circular]";
   }
 }
 
+export type LogLevel = "debug" | "info" | "warn" | "error";
+
+export interface LoggerOptions {
+  level: LogLevel;
+}
+
+const PREFIX = "✦Aj";
+
 function getTimeLabel(label: string) {
-  return `✦Aj Latency ${label}`;
+  return `${PREFIX} LATENCY ${label}`;
+}
+
+function getMessage(obj: unknown, msg: unknown, args: unknown[]) {
+  // The first argument was the message
+  if (typeof obj === "string") {
+    return format(obj, ...args);
+  }
+
+  // The second argument was the message
+  if (typeof msg === "string") {
+    return format(msg, ...args);
+  }
+
+  if (
+    typeof obj === "object" &&
+    obj !== null &&
+    "msg" in obj &&
+    typeof obj.msg === "string"
+  ) {
+    return format(obj.msg, ...args);
+  }
+}
+
+function getOutput(obj: unknown, msg: unknown, args: unknown[]) {
+  let output = getMessage(obj, msg, args);
+  if (typeof output !== "string") {
+    return;
+  }
+
+  if (typeof obj === "object" && obj !== null) {
+    for (const [key, value] of Object.entries(obj)) {
+      output += `\n    ${key}: ${tryStringify(value)}`;
```

Comment:
```suggestion
      output += `\n✧    ${key}: ${tryStringify(value)}`;
```
@morekid like this?

Reply:
We keep the examples in the `examples/` directory in this repository so they can be tested with the workspace changes. Here's the output from one of them:
```
✦Aj DEBUG Decide response
✧    id: "req_01hzpvr7nkftn9ct33a62q0vrz"
✧    fingerprint: "fp_1_3e48ef9d22e096da6838540fb846999890462c8a32730a4f7a5eaee6945315f7"
✧    path: "/api/arcjet"
✧    runtime: "node"
✧    ttl: 0
✧    conclusion: "ALLOW"
✧    reason: {"type":"SHIELD","shieldTriggered":false}
✧    ruleResults: [{"ruleId":"","ttl":0,"state":"RUN","conclusion":"ALLOW","reason":{"type":"SHIELD","shieldTriggered":false}}]
✦Aj LATENCY decideApi: 88.984ms
✦Aj LATENCY remote: 89.027ms
```

Reply:
We're going to avoid this right now because it is not a multi-line text description but instead properties from the structured logger.